brainscore/metrics/regression.py

```diff
@@ -69,3 +90,25 @@ def linear_regression(xarray_kwargs=None):
 def pearsonr_correlation(xarray_kwargs=None):
     xarray_kwargs = xarray_kwargs or {}
     return XarrayCorrelation(scipy.stats.pearsonr, **xarray_kwargs)
+
+
+def single_regression(xarray_kwargs=None):
+    regression = SingleRegression()
+    xarray_kwargs = xarray_kwargs or {}
+    regression = XarrayRegression(regression, **xarray_kwargs)
+    return regression
+
+
+def pearsonr(x, y):
+    xmean = x.mean(axis=0, keepdims=True)
+    ymean = y.mean(axis=0, keepdims=True)
+
+    xm = x - xmean
+    ym = y - ymean
+
+    normxm = scipy.linalg.norm(xm, axis=0, keepdims=True)
+    normym = scipy.linalg.norm(ym, axis=0, keepdims=True)
+
+    r = ((xm/normxm)*(ym/normym)).sum(axis=0)
+
+    return r
```

Comment:
is this different from `scipy.stats.pearsonr`?

Reply:
It allows x to have multiple columns and gives the pearsonr columnwise x with y. Scipy only works with vectors. This makes it much faster for the single neuron regression mapping.